### PR TITLE
Workspace settings page updates

### PR DESCRIPTION
**Pull request by siovan-ssc**
Added functionality to WorkspaceSettingsPage:
- Workspace leads can update workspace budgets and delete workspaces
- Admins can see the settings but not make any changes (fields are disabled)
- Other user types cannot see these settings

Updated DatahubAuthView so workspace leads are also required to have a project acronym specified.

Added basic .feature file for testing of WorkspaceSettingsPage.

### CODE REVIEW

File: Portal/src/Datahub.Portal/Pages/Workspace/Settings/WorkspaceSettingsPage.razor [hunk 1]
```diff
@@ -51,16 +86,46 @@
...
+
+        var currentUser = await _userInformationService.GetCurrentPortalUserAsync();
+        var _projectUsers = await _projectUserManagementService.GetProjectUsersAsync(WorkspaceAcronym);
+
+        foreach (var user in _projectUsers)
+        {
+            if (user.Role.Id.Equals(2) && user.PortalUser.Id.Equals(currentUser.Id))
+            {
+                _isDisabled = false;
+            }
```
Comment (yjmrobert) [inline Portal/src/Datahub.Portal/Pages/Workspace/Settings/WorkspaceSettingsPage.razor]: That's the right idea, but I'll show you the `NotAuthorized` section in the Auth view that'll simplify things

**Review by yjmrobert — CHANGES_REQUESTED** [review]
Nice work, I like the update in progress UX! Do you happen to have a screenshot of it?


Let's update the disabled to work off the user roles via the NotAuthorized render component.

All that's left is to send the new Workspace Definition onto the queue. You might be able to reuse a method in one of the project management services. 

I'm more than happy to jump on a call to explain some of the topology as there are a lot of moving parts

**Comment by siovan-ssc** [comment]
Updated WorkspaceSettingsPage:
- Disabled works off Authorized / NotAuthorized components
- Added _requestManagementService await for sending to the queue

Also, here is what the UX looks like:
When budget has been changed:
Screenshot 2024-03-11 153819

When saving the new budget:
Screenshot 2024-03-11 153804

**Comment by yjmrobert** [comment]
Looks good! Great work on the UI/UX. I did a minor update to only allow datahub admins instead of workspace leads (just until we figure out how costing is going to work in production).

Cheers!

**Pull request merged**